src/v/config/configuration.cc

```diff
@@ -1221,15 +1221,15 @@ configuration::configuration()
       "kafka_nodelete_topics",
       "Prevents the topics in the list from being deleted via the kafka api",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
-      {"__audit", "__consumer_offsets", "_schemas"},
+      {"__audit_log", "__consumer_offsets", "_schemas"},
       &validate_non_empty_string_vec)
   , kafka_noproduce_topics(
       *this,
       "kafka_noproduce_topics",
       "Prevents the topics in the list from having message produced to them "
       "via the kafka api",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
-      {"__audit"},
+      {"__audit_log"},
```

Comment:
We already have logic to prevent users from not producing to the audit log in `produce.cc`. We shouldn't depend on `__audit_log` existing here anyway because the user could just modify this and remove the audit log from the list of noproduce topics. I think the default value for this config can be an empty list now.

Reply:
Actually after I just pushed the change to remove it, I think the argument for leaving it in here is the same for why to leave it in nodelete.  If auditing is disabled, I think the user should expect to jump through an additional hoop before making modifications to the topic.

Reply:
If thats true shouldn't be that way. We should make a patch to prevent create_topics from succeeding if `__audit_log` is the topic name